### ISSUE TITLE
caching.New(...) should take a Monitor and count cache hits

### ISSUE THREAD

**Issue by jonasfj** [taskcluster/taskcluster-worker#334]
`caching.New(...)` should additionally accept a `runtime.Monitor` instance...

So that setup looks somewhat like:
```go
cache := caching.New(constructor, true || false, garbageCollector, monitor.prefix('<prefix>'))
```

Then the `cache` should track cache hit/miss ratio, as well as constructor duration... as metrics: `<prefix>.cache-hit` and `<prefix>.cache-miss` and `<prefix>.creation-time`

That way if we pass `monitor.prefix('image-manager')` to `caching.New(...)` we get metrics: 
`image-manager.cache-hit` and `image-manager.cache-miss` and `image-manager.creation-time` which would be pretty neat :)

**Comment by Biboswan** [comment]
is this issue is available to be taken ?

**Comment by jonasfj** [comment]
sure, go for it... note that we're going to merge PR #343 soon which adds new usages of `caching` :)

**Comment by jonasfj** [comment]
Fixed in #344
